test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+                expect(_.floor.apply(_, [capacity(2.2,3.3,4.4), 2])).to.eql(31.94);
+            });
+        });
+
+        describe('max', function () {
+
+            var params;
+
+            beforeEach(function () {
+                params = datasets.max();
+            });
+
+            it('should find max value (5) in user array', function () {
+                expect(params).to.have.length(1);
+
+                for(var i=0; i<params[0].length; i++){
+                    expect(typeof params[0][i]).to.eql('number');
+                }
+
+                expect(_.max.apply(_, datasets.max(params))).to.eql(5);
+            });
+        });
+
+
+        describe('maxBy', function () {
+
+            describe('maxBy - array', function () {
+
+                var params, arr1, arr2;
+
+                beforeEach(function () {
+                    params = datasets.maxBy1();
+                    arr1 = [6, 5, 4, 8, 3];
+                    arr2 = [-4, -2, -6, 0];
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(arr1 instanceof Array && arr2 instanceof Array).to.eql(true);
+                });
+
+                it('should return maxBy for array', function () {
+                    expect(_.maxBy.apply(_, datasets.maxBy1(arr1))).to.eql(8);
+                    expect(_.maxBy.apply(_, datasets.maxBy1(arr2))).to.eql(0);
+                });
+            });
+
+            /*
+            describe('maxBy - function', function () {
+                it('should use maxBy for function', function () {
+
+                });
+            });
+            //I have no idea how to do it. Probably maxBy doesn't support function. Help!
+            */
+
+            describe('maxBy - object', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.maxBy3();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(2);
+                    expect(params[0] instanceof Object).to.eql(true);
+                    expect(typeof params[1]).to.eql('string');
+                });
+
+                it('should use maxBy object', function () {
+                    expect(_.maxBy.apply(_, params)).to.eql({ Name: 'Edwin', Gold: 2500 });
+                });
+            });
+
+            describe('maxBy - string', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.maxBy4();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                });
+                
+                it('should find max value in string', function () {
+                    expect(_.maxBy.apply(_, params)).to.eql('word');
```

Comment:
To be honest, I don't know what to say. From one point you are really testing something that even I didn't know that will happen in that case, but on the other hand test case is completely wrong as I mentioned above.

What I want to ask is, how did you get information how lodash would behave in that case? If you could provide me with the source that would be great.